src/edu/washington/escience/myria/operator/Split.java

```diff
@@ -0,0 +1,146 @@
+package edu.washington.escience.myria.operator;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+
+import edu.washington.escience.myria.DbException;
+import edu.washington.escience.myria.Schema;
+import edu.washington.escience.myria.Type;
+import edu.washington.escience.myria.column.Column;
+import edu.washington.escience.myria.column.builder.ColumnBuilder;
+import edu.washington.escience.myria.column.builder.ColumnFactory;
+import edu.washington.escience.myria.storage.TupleBatch;
+import edu.washington.escience.myria.storage.TupleUtils;
+
+/**
+ * Generic apply operator.
+ */
+public class Split extends UnaryOperator {
+  /***/
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * Name of column to split on regex. Must have String type.
+   */
+  private final String columnName;
```

Comment:
We tend towards using indices rather than column names for this purpose. The reason is that the optimizer refers to everything by position and there are often columns with unimportant/autogenerated names at intermediate nodes in the query plan.